System/Files/file.cpp

```diff
@@ -418,3 +418,81 @@ void
 string_save (string s, url u) {
   (void) save_string (u, s, false);
 }
+
+bool
+append_string (url u, string s, bool fatal) {
+  if (!is_local_and_single (u)) {
+    cerr << "Failed to load url: [" << as_string (u) << "]" << LF;
+    return file_failure (fatal, "Must be a local and single file");
+  }
+
+  url  u_iter  = expand (u);
+  url  u_target= url_none ();
+  bool err     = false;
```

Comment:
Please return directly when failure instead of toggling or reading this variable.